app/models/miq_expression.rb

```diff
@@ -1593,6 +1593,31 @@ def self._quick_search?(e)
     end
   end
 
+  def fields
+    self.class._fields(exp)
+  end
+
+  def self._fields(expression)
```

Comment:
@lpichler looks better, but I think there are a few cases that might catch you out here:

It looks like it's assuming here that if the hash doesn't have a `field` key, then it must be a composite expression. That's not necessarily so - it's quite possible to have an expression that doesn't have a `field` in it all (specifically - ones where `tag` is used instead).

Building on that - I wonder is it right to have an instance method `fields` for an object that might not have any?